presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1104,34 +1107,32 @@ else if (criteria instanceof JoinOn) {
 
             for (Expression conjunct : ExpressionUtils.extractConjuncts((Expression) optimizedExpression)) {
                 conjunct = ExpressionUtils.normalize(conjunct);
```

Comment:
And one more comment on keeping the specific code for any comparison operator (not only equality). 

In case when expression corresponding to right (replicated) table is expensive to compute (CPU-wise) - with current approach we may be more efficient. Now computation is done once, on machine which executes right relation. If we always used postInnerJoin it would be computed multiple times for same rows - on machines performing JOIN.

Just my 2 cents. Probably not very practical scenario.